File: src/main/java/tw/shounenwind/plurkconnection/PlurkConnectionException.java
package tw.shounenwind.plurkconnection;

import tw.shounenwind.plurkconnection.responses.IResponse;

public class PlurkConnectionException extends Exception {

    private final IResponse apiResponse;

    public PlurkConnectionException(IResponse response) {
        super(response.toString());
        this.apiResponse = response;
    }

    public PlurkConnectionException(IResponse response, Exception e) {
        super(response.getStatusCode() + ": " + response.toString(), e);
        this.apiResponse = response;
    }

    public IResponse getApiResponseString() {
        return apiResponse;
    }

    @Override
    public void printStackTrace() {
        super.printStackTrace();
    }
}

File: library/src/main/java/tw/shounenwind/plurkconnection/PlurkConnectionException.kt
package tw.shounenwind.plurkconnection

class PlurkConnectionException : Exception {


    constructor(target: String) : super(target)
    constructor(target: String, message: String) : super("$target, $message")
    constructor(target: String, message: String, cause: Throwable) : super("$target, $message", cause)
    constructor(target: String, cause: Throwable) : super(target, cause)


}